### PR TITLE
Add doxygen-style comments

### LINKED ISSUE #95: Improve documentation in the contracts, use doxygen style
https://github.com/ethereum/wiki/wiki/Ethereum-Natural-Specification-Format and also check out contracts from other projects

### PR DESCRIPTION
**Pull request by nberger**
Fixes #95

### CODE REVIEW

File: contracts/LifToken.sol [hunk 1]
```diff
@@ -3,6 +3,13 @@ pragma solidity ^0.4.13;
 import "zeppelin-solidity/contracts/token/MintableToken.sol";
 import "zeppelin-solidity/contracts/lifecycle/Pausable.sol";
 
+/**
+ * @title Líf, the Winding Tree token
+ *
+ * Implementation of Líf, the ERC20 token for Winding Tree, with extra methods
+ * for transfers with contract calls.
```
Comment (AugustoL) [inline contracts/LifToken.sol]: what about "to transfer value and data allowing the execution of calls in transfer" ? or smth like that

File: contracts/LifToken.sol [hunk 2]
```diff
@@ -17,20 +24,14 @@ contract LifToken is MintableToken, Pausable {
...
-
-  function transferFrom(address _from, address _to, uint256 _value) whenNotPaused returns (bool) {
-    return super.transferFrom(_from, _to, _value);
-  }
-
-  // approveData is an addition to ERC20 token methods. It allows approving the transference of value and execute a data call on the approval transaction
-  function approveData(address spender, uint value, bytes data) whenNotPaused {
+  /**
+   * @dev `approveData` is an addition to ERC20 token methods.
+   * It allows to approve the transfer of value and execute a call with data in the same transaction
```
Comment (AugustoL) [inline contracts/LifToken.sol]: It allows to approve the transfer of value and execute a call with the data sent

File: contracts/LifToken.sol [hunk 3]
```diff
@@ -41,8 +42,14 @@ contract LifToken is MintableToken, Pausable {
 
   }
 
-  // transferData is an addition to ERC20 token methods. It allows to transfer value and data on each transaction
-  function transferData(address to, uint value, bytes data) whenNotPaused {
+  /**
+   * @dev Addition to ERC20 token methods. Transfer tokens to a specified address and make a contract call
```
Comment (AugustoL) [inline contracts/LifToken.sol]: Transfer tokens to a specified address and execute a call with the data sent

File: contracts/LifToken.sol (hunk 1 above)
Comment (nberger) [reply contracts/LifToken.sol]: Even though I like "contract call" or "function call" more, I'm fine with "execution of calls". However, it should be singular like in "execution of a call on transfer", right? It's a single call, not multiple calls

Comment (AugustoL) [reply contracts/LifToken.sol]: yes, singular.

File: contracts/LifCrowdsale.sol [hunk 4]
```diff
@@ -6,78 +6,119 @@ import "zeppelin-solidity/contracts/math/SafeMath.sol";
...
+ */
 contract LifCrowdsale is Ownable, Pausable {
   using SafeMath for uint256;
 
-  // The token being sold
+  // The token being sold.
   LifToken public token;
 
-  // start and end timestamps of the public presale
+  // start and end timestamps of the public presale.
```
Comment (AugustoL) [inline contracts/LifCrowdsale.sol]: It shouldn't start with uppercase here like all the comments?

Comment (AugustoL) [inline contracts/LifCrowdsale.sol]: It shouldn't start with uppercase here like all the comments?

File: contracts/LifCrowdsale.sol [hunk 5]
```diff
@@ -6,78 +6,119 @@ import "zeppelin-solidity/contracts/math/SafeMath.sol";
...
-  // start and end timestamps where investments are allowed (both inclusive)
+  // Beginning of the period where tokens can be purchased at rate `rate1`.
   uint256 public startTimestamp;
+  // Moment after which the rate to buy tokens goes from `rate1` to `rate2`.
   uint256 public end1Timestamp;
+  // Marks the end of the Token Generation Event.
   uint256 public end2Timestamp;
 
-  // address where funds are collected
+  // address of the Winding Tree Foundation wallet. Funds up to the soft cap are
```
Comment (AugustoL) [inline contracts/LifCrowdsale.sol]: start with uppercase here too?

**Comment by nberger** [comment]
What style of block do we like the most, this one:

```
  /**
   * @dev Addition to ERC20 token methods. Transfer tokens from one address to another and make a contract call
   * on the same transaction
   *
   * @param from The address which you want to send tokens from
   * @param to The address which you want to transfer to
   * @param value The amout of tokens to be transferred
   * @param data ABI-encoded contract call. For example generated using web3's getData method
   */
```

Or this one:
```
  /**
     @dev Addition to ERC20 token methods. Transfer tokens from one address to another and make a contract call
     on the same transaction

     @param from The address which you want to send tokens from
     @param to The address which you want to transfer to
     @param value The amout of tokens to be transferred
     @param data ABI-encoded contract call. For example generated using web3's getData method
   */
```

I like the second better

**Comment by AugustoL** [comment]
The second :)

**Pull request merged**